### PR TITLE
buildFHSEnvBubblewrap: escape runScript path

### LINKED ISSUE #230651: buildFHSUserEnv fails to run for paths with spaces
### Describe the bug
`buildFHSUserEnv` fails to run for paths with spaces.
See: https://github.com/thiagokokada/nix-alien/issues/71

### Steps To Reproduce
Run `nix-alien` on path with spaces

### Expected behavior
Works correctly

### Additional context
This kind of file was generated by `buildFHSUserEnv`
```
#!/nix/store/rnkas52f8868g1hjdlldbvh6snm3pglv-bash-5.2-p15/bin/bash
source /etc/profile
cat > /etc/ld.so.conf <<EOF
/lib
/lib/x86_64-linux-gnu
/lib64
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib64
/lib/i386-linux-gnu
/lib32
/usr/lib/i386-linux-gnu
/usr/lib32
/run/opengl-driver/lib
/run/opengl-driver-32/lib
EOF
ldconfig &> /dev/null

exec /path/to/file with spaces "$@"
```

### Notify maintainers

@Atemu 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.26, NixOS, 23.05 (Stoat), 23.05.20230429.08e4dc3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-21.11.336297.47cd6702934"`
 - nixpkgs: `/nix/store/f9npl9988fipf059n1dh2s6gb6pbfc0i-9c0slwp2dyc6z2gxkcyrqxifbik0n68d-source`
```

nixpkgs in flake.lock
```
    "nixpkgs": {
      "locked": {
        "lastModified": 1682786779,
        "narHash": "sha256-m7QFzPS/CE8hbkbIVK4UStihAQMtczr0vSpOgETOM1g=",
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "08e4dc3a907a6dfec8bb3bbf1540d8abbffea22b",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
```

### PR DESCRIPTION
**Pull request by Atemu**
Fixes https://github.com/NixOS/nixpkgs/issues/230651

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Atemu** [comment]
cc for testing @hissssst

File: pkgs/build-support/build-fhsenv-bubblewrap/default.nix [hunk 1]
```diff
@@ -111,7 +111,7 @@ let
   init = run: writeShellScript "${name}-init" ''
     source /etc/profile
     ${createLdConfCache}
-    exec ${run} "$@"
+    exec "${run}" "$@"
```
Comment (wegank) [inline pkgs/build-support/build-fhsenv-bubblewrap/default.nix:114]:
```suggestion
    exec ${lib.escapeShellArg run} "$@"
```

**Comment by hissssst** [comment]
Works correctly

**Comment by ElXreno** [comment]
This PR breaks the `lens` package:
```
/nix/store/sayzkznm9n8knqmg5kdp1bwzxck8nfxb-lens-6.4.15-init: line 19: /home/elxreno/appimage-exec.sh -w /nix/store/53g61qbaqqskcwq4hbclgdrv4ik0p89m-lens-6.4.15-extracted --: No such file or directory
```
I've got this PR by bisecting the good and bad commits.

**Pull request merged**